src/components/button/button.tsx

```diff
@@ -0,0 +1,30 @@
+import cn from "classnames";
+import Image from "next/image";
+
+interface ButtonProps {
+    label: string;
+    icon?: string;
+    fill: boolean;
+    onClick?: () => void;
+    type?: "submit";
+}
```

Comment:
I suggest you to use `type` instead of `interface`.

Also, there's no need to export this type.

I suggest to use children prop instead of label.
See the difference:

label:
```tsx
<Button label="Download" />
```
children:
```tsx
<Button>Download</Button>
```

There's a type that you can use to copy all props from `<button>`

```tsx
type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  icon?: string
}

const Button = ({ type = "button", icon, className, children, ...restProps }) => {
  return (
    <button
      {...restProps}
      type={type}
      className={clsx("...", className)}
    >
      {children}
    </button>
}
```